CI-DNS.ign

```diff
@@ -0,0 +1 @@
0%20%20%20%20%20SERIAL_NUMBER%20%2B%3D%201%0A%20%20%20%20%20%20%20%20%20%20%20%20db%20%3D%20build_coredb_file(DOMAIN%2C%20SERIAL_NUMBER%2C%20ENTRIES)%0A%20%20%20%20%20%20%20%20%20%20%20%20print%20%22Writing%20version%20%7B%7D%20to%20file%20%7B%7D%22.format(SERIAL_NUMBER%2C%20COREDB_FILE)%0A%20%20%20%20%20%20%20%20%20%20%20%20with%20open(COREDB_FILE%2C%20'w')%20as%20f%3A%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20f.write(db)%0A%20%20%20%20%20%20%20%20else%3A%0A%20%20%20%20%20%20%20%20%20%20%20%20print%20%22Invalid%20format%20of%20body%22%2C%20repr(body)%0A%0A%0Aif%20__name__%20%3D%3D%20'__main__'%3A%0A%20%20%20%20Handler%20%3D%20ServerHandler%0A%20%20%20%20httpd%20%3D%20SocketServer.TCPServer((%22%22%2C%20PORT)%2C%20Handler)%0A%20%20%20%20print%20%22serving%20at%20port%22%2C%20PORT%0A%20%20%20%20httpd.serve_forever()%0A","verification":{}},"mode":420},{"filesystem":"root","path":"/etc/coredns/Corefile","contents":{"source":"data:,.%20%7B%0A%20%20%20%20log%0A%20%20%20%20errors%0A%0A%20%20%20%20forward%20.%20%2Fetc%2Fresolv.conf%20%7B%0A%20%20%20%20%20%20%20%20except%20shiftstack.ci%0A%20%20%20%20%7D%0A%7D%0A%0Ashiftstack.ci%20%7B%0A%20%20%20%20log%0A%20%20%20%20errors%0A%0A%20%20%20%20file%20%2Fetc%2Fcoredns%2Fdb.shiftstack.ci%20%7B%0A%20%20%20%20%20%20%20%20reload%2010s%0A%20%20%20%20%7D%0A%7D%0A","verification":{}},"mode":420},{"filesystem":"root","path":"/etc/coredns/db.shiftstack.ci","contents":{"source":"data:,%24ORIGIN%20shiftstack.ci.%0A%40%20%20%20%203600%20IN%20SOA%20dns.shiftstack.ci.%20hostmaster%20(%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%202017042752%20%3B%20serial%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%207200%20%20%20%20%20%20%20%3B%20refresh%20(2%20hours)%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%203600%20%20%20%20%20%20%20%3B%20retry%20(1%20hour)%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%201209600%20%20%20%20%3B%20expire%20(2%20weeks)%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%203600%20%20%20%20%20%20%20%3B%20minimum%20(1%20hour)%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20)%0A","verification":{}},"mode":420}]},"systemd":{"units":[{"contents":"[Unit]\nDescription=CI DNS\n\n[Service]\nExecStart=/bin/podman run --rm -i -t -m 128m --net host --cap-add=NET_ADMIN -v /etc/coredns:/etc/coredns:Z openshift/origin-coredns:v4.0 -conf /etc/coredns/Corefile\nRestart=always\nRestartSec=10\n\n[Install]\nWantedBy=multi-user.target\n","enabled":true,"name":"ci-dns.service"},{"contents":"[Unit]\nDescription=Dynamic DNS updater\n\n[Service]\nExecStart=/bin/python /etc/coredns/dynamic-update.py\nRestart=always\nRestartSec=10\n\n[Install]\nWantedBy=multi-user.target\n","enabled":true,"name":"dynamic-dns-updater.service"}]}}
```

Comment:
Same comment about the ssh key.

Reply:
This file is actually just generated from the human readable YAML. We might want a lint job that checks changes to the YAML are correctly generated in the .ign file. It depends how often we need to make any changes to this though.

FWIW, I have been using this ignition config unchanged for a few months. If we start adding other CI services (monitoring/log aggregation/cleanup/etc.) to this VM though, we do have to be careful about making sure the .yaml and .ign stay in sync. The ideal workflow to (re)create this VM is to git checkout master and use the .ign file.

Reply:
Is there an easy way to generate the ignition file from the yaml config? If so, perhaps there is no need to store it in git and can be documented instead?